lambdas/data-migration1/src/rules.ts

```diff
@@ -0,0 +1,194 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { PostgresRuleRecord } from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordDoesNotExist } from '@cumulus/errors';
+import { RecordAlreadyMigrated } from './errors';
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/rules' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+/**
+ *
+ * Retrieve cumulus_id from Provider using id.
+ *
+ * @param {string} name - Provider ID
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordDoesNotExist} if record cannot be found
+const getProviderCumulusId = async (
+  name: string,
+  knex: Knex
+): Promise<any> => {
+  const record = await knex.select(knex.ref('cumulus_id').as('provider_cumulus_id'))
+    .from<PostgresProviderRecord>('providers')
+    .where({ name: name })
+    .first();
+
+  if (record === undefined) {
+    throw new RecordDoesNotExist(`Provider with id ${name} does not exist.`);
+  }
+  return record.provider_cumulus_id;
+};
+ */
+
+/**
+ *
+ * Retrieve cumulus_id from Collection using name and version.
+ *
+ * @param {string} name - Collection Name
+ * @param {string} version - Collection version
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordDoesNotExist} if record cannot be found
+const getCollectionCumulusId = async (
+  name : string,
+  version: string,
+  knex: Knex
+): Promise<number> => {
+  const record = await knex.select(knex.ref('cumulus_id').as('collection_cumulus_id'))
+    .from<PostgresCollectionRecord>('collections')
+    .where({ name: name, version: version })
+    .first();
+
+  if (record === undefined) {
+    throw new RecordDoesNotExist(`Collection with name ${name} and version ${version} does not exist.`);
+  }
+  return record.collection_cumulus_id;
+};
+*/
+
+/**
```

Comment:
Below is the generalized `getCumulusId` function I wrote to contrast the two non generalized functions above. My main issue is that I'm not sure how to pass in a type so that I could actually query the tables like:
```
const record = await knex.select('cumulus_id')
    .from<TYPE>(table)
    .where(whereClause)
    .first();
```
Any input on how to do that in TypeScript would be much appreciated (here's a cookie 🍪 )